pkg/controller/templateupdaterequest/templateupdaterequest_controller.go

```diff
@@ -0,0 +1,315 @@
+func (r *ReconcileTemplateUpdateRequest) checkTransitionTimeAndDelete(logger logr.Logger, tur *toolchainv1alpha1.TemplateUpdateRequest, completeCond toolchainv1alpha1.Condition) (bool, time.Duration, error) {
+	logger.Info("the ChangeTierRequest is completed so we can deal with its deletion")
+	timeSinceCompletion := time.Since(completeCond.LastTransitionTime.Time)
+	if timeSinceCompletion >= DeletionTimeout {
+		logger.Info("the TemplateUpdateRequest has been completes for a long enough time, so it's ready to be deleted")
+		if err := r.client.Delete(context.TODO(), tur); err != nil {
+			return false, 0, errs.Wrapf(err, "unable to delete the TemplateUpdateRequest resource '%s'", tur.Name)
+		}
+		return true, 0, nil
+	}
+	diff := DeletionTimeout - timeSinceCompletion
+	logger.Info("the TemplateUpdateRequest has been completed for short time, so it's not going to be deleted yet", "reconcileAfter", diff.String())
+	return false, diff, nil
+}
+
+// --------------------------------------------------
+// status updates
+// --------------------------------------------------
+
+func toFailure(err error) toolchainv1alpha1.Condition {
+	return toolchainv1alpha1.Condition{
+		Type:    toolchainv1alpha1.TemplateUpdateRequestComplete,
+		Status:  corev1.ConditionFalse,
+		Reason:  toolchainv1alpha1.TemplateUpdateRequestUnableToUpdateReason,
+		Message: err.Error(),
+	}
+}
+
+func toBeUpdating() toolchainv1alpha1.Condition {
+	return toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.TemplateUpdateRequestComplete,
+		Status: corev1.ConditionFalse,
+		Reason: toolchainv1alpha1.TemplateUpdateRequestUpdatingReason,
+	}
+}
+
+func toBeComplete() toolchainv1alpha1.Condition {
+	return toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.TemplateUpdateRequestComplete,
+		Status: corev1.ConditionTrue,
+		Reason: toolchainv1alpha1.TemplateUpdateRequestUpdatedReason,
+	}
+}
+
+// updateStatusConditions updates the TemplateUpdateRequest status conditions with the new conditions
+func (r *ReconcileTemplateUpdateRequest) updateStatusConditions(logger logr.Logger, tur *toolchainv1alpha1.TemplateUpdateRequest, newConditions ...toolchainv1alpha1.Condition) error {
+	var updated bool
+	tur.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(tur.Status.Conditions, newConditions...)
+	if !updated {
+		// Nothing changed
+		logger.Info("TemplateUpdateRequest status conditions unchanged")
+		return nil
+	}
+	return r.client.Status().Update(context.TODO(), tur)
+}
+
+// setStatusConditionsUpdating sets the TemplateUpdateRequest status condition to `complete=false/reason=updating` and retains the sync indexes
+func (r *ReconcileTemplateUpdateRequest) setStatusConditionsUpdating(logger logr.Logger, tur *toolchainv1alpha1.TemplateUpdateRequest, syncIndexes map[string]string, conditions ...toolchainv1alpha1.Condition) error {
+	tur.Status.SyncIndexes = syncIndexes
+	tur.Status.Conditions = []toolchainv1alpha1.Condition{toBeUpdating()}
```

Comment:
I found why the e2e tests are failing - you don't use the `condition.AddOrUpdateStatusConditions`  function here, so it doesn't set the `lastTransitionTime`. The e2e tests have error:
```
{"level":"error","ts":1593513373.3354383,"logger":"controller_templateupdaterequest","msg":"Unable to update the TemplateUpdateRequest status","Request.Namespace":"codeready-toolchain-host-30102747","Request.Name":"cheesecakelover07","error":"TemplateUpdateRequest.toolchain.dev.openshift.com \"cheesecakelover07\" is invalid: status.conditions.lastTransitionTime: Invalid value: \"null\": status.conditions.lastTransitionTime in body must be of type string: \"null\"","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/codeready-toolchain/host-operator/pkg/controller/templateupdaterequest.(*ReconcileTemplateUpdateRequest).Reconcile\n\t/tmp/codeready-toolchain/host-operator/pkg/controller/templateupdaterequest/templateupdaterequest_controller.go:142\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:88"}
```
see https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/codeready-toolchain_toolchain-e2e/127/pull-ci-codeready-toolchain-toolchain-e2e-master-e2e/1277879736474800128#1:build-log.txt%3A7113

my guess is that it should look like this:
```suggestion
	tur.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(tur.Status.Conditions, toBeUpdating())
```
but I haven't had a chance to verify it locally

Reply:
I just verified locally and you were right 💪 🙌 
The controller failed to perform the update, and this is the kind of error that were found in the logs:
```
status.conditions.lastTransitionTime: Invalid value: "null"
```